src/app/shared/project-form/project-form.component.ts

```diff
@@ -0,0 +1,64 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { Router } from '@angular/router';
+import { Subscription } from 'rxjs';
+import { FormServiceService } from '../services/form-service.service';
+
+
+@Component({
+  selector: 'app-project-form',
+  templateUrl: './project-form.component.html',
+  styleUrls: ['./project-form.component.css']
+})
+export class ProjectFormComponent implements OnInit {
+
+  slider: string = '0';
+  subscription: Subscription;
+  buttonText: string;
+  showSlider = false;
+  projectForm: FormGroup;
+  
+  constructor(private formService: FormServiceService, private router: Router) {}
+
+  ngOnInit(): void {
+    if(String(this.router.url).toLocaleLowerCase().includes('edit')){
+      this.buttonText = 'Update Project';
+      this.showSlider = true;
+    }
+    else {
+      this.buttonText = 'Create Project';
+      this.showSlider = false;
+    }
+
+    this.onInitForm()
+  }
+
+  private onInitForm(){
+    let projectName = '';
+    let clientName = '';
+    let startDate = '';
+    let endDate = '';
+    let range = '';
+    let description = '';
+
+    this.projectForm = new FormGroup({
+      'projectName': new FormControl(projectName, Validators.required),
```

Comment:
Can we do like this, 'projectName': new FormControl('', Validators.required), instead of declaring variables like, let projectName = ''; as we are not using these variables anywhere in the method.

Reply:
Those variables will be used later in the project.